src/IndexingMsg.chpl

```diff
@@ -480,6 +502,107 @@ module IndexingMsg
             when (DType.Bool, DType.Bool) {
                 return ivBoolHelper(bool);
             }
+            when (DType.BigInt, DType.Int64) {
+                var e = toSymEntry(gX,bigint);
+                var iv = toSymEntry(gIV,int);
+                if (e.size == 0) && (iv.size == 0) {
+                    var a = st.addEntry(rname, new shared SymEntry(makeDistArray(0, bigint)));
+                    var repMsg = "created " + st.attrib(rname);
+                    imLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),repMsg);
+                    return new MsgTuple(repMsg, MsgType.NORMAL);
+                }
+                var ivMin = min reduce iv.a;
+                var ivMax = max reduce iv.a;
+                if ivMin < 0 {
+                    var errorMsg = "Error: %s: OOBindex %i < 0".format(pn,ivMin);
+                    imLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+                    return new MsgTuple(errorMsg,MsgType.ERROR);
+                }
+                if ivMax >= e.size {
+                    var errorMsg = "Error: %s: OOBindex %i > %i".format(pn,ivMin,e.size-1);
+                    imLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+                    return new MsgTuple(errorMsg,MsgType.ERROR);
+                }
+                var tmp = makeDistArray(iv.size, bigint);
+                ref a2 = e.a;
+                ref iva = iv.a;
+                forall (t,idx) in zip(tmp,iva) {
+                    // TODO find a way to use aggregators
```

Comment:
Should we add an issue to research this?

Reply:
Added https://github.com/Bears-R-Us/arkouda/issues/1994